node/src/syncer.rs

```diff
@@ -443,6 +445,19 @@ where
             return;
         }
 
+        let syncing_window_start = Time::now().checked_sub(SYNCING_WINDOW).unwrap_or_else(|| {
+            warn!("underflow when computing syncing window start, defaulting to unix epoch");
+            Time::unix_epoch()
+        });
+
+        // make sure we're inside the syncing window before we start
+        if let Ok(known_header) = self.store.get_by_height(next_batch.end() + 1).await {
+            if known_header.time().before(syncing_window_start) {
+                debug!("synced to the end of syncing window");
```

Comment:
i think when we reach this line, we can safely say that this batch and anything before it, should not be rechecked when `fetch_next_batch` is called.

we should introduce a height limit that will be set to `next_batch.end() + 1` and `calculate_range_to_fetch` should exclude anything before that.

Reply:
Caching added a bit of code here, but added saing the known syncing window edge and accounting for it in ranges calculcation 👍